ArticleTemplates/assets/js/modules/cards.js

```diff
@@ -1,96 +1,45 @@
 define([
-    'flipSnap',
     'modules/util'
-],
+], 
 function (
-    flipSnap,
     util
 ) {
     'use strict';
 
     var initialised = false;
+    var observer = null;
 
-    function setupGlobals() {
-        window.articleCardsInserter = articleCardsInserter;
-        window.articleCardsFailed = articleCardsFailed;
-
-        window.applyNativeFunctionCall('articleCardsInserter');
-        window.applyNativeFunctionCall('articleCardsFailed');
-    }
-
-    function articleCardsInserter(html) {
-        var relatedContentWrapper = document.getElementsByClassName('related-content__wrapper')[0];
-
-        if (relatedContentWrapper) {
-            if (!html) {
-                articleCardsFailed();
-            } else {
-                relatedContentWrapper.innerHTML = html;
-                snapToGrid();
-            }
+    function ready(config) {
+        if (!initialised) {
+            initialised = true;
+            setupGlobals();
         }
     }
 
-    function articleCardsFailed() {
-        var relatedContent = document.getElementsByClassName('related-content')[0];
-
+    function getRelatedContentPosition() {
+        var relatedContent = document.querySelector('.related-content');
         if (relatedContent) {
-            relatedContent.classList.add('related-content--has-failed');       
-        }
-    }
-
-    function snapToGrid() {
-        var relatedContentList = document.getElementsByClassName('related-content__list')[0];
-
-        if (relatedContentList) {
-            setUpFlipSnap(relatedContentList);
-            window.addEventListener('resize', util.debounce(onResize.bind(null, relatedContentList), 100));
-        }
-    }
-
-    function onResize(relatedContentList) {
-        if (flipSnap && flipSnap.destroy && relatedContentList) {
-            flipSnap.destroy();
-            relatedContentList.removeAttribute('style');
-            setUpFlipSnap(relatedContentList);
+            observer.observe(relatedContent, { attributes: true });
```

Comment:
Do we need to set up an new `observer.observe` everytime here? Is setting it up once on the body `observer.observe(window.document.body, { ... })` not enough?